File: Pset 1/cash/cash.c
// Stela Gavrilova
// 16 Feb 2019
// Problem Set 1

// A program that first asks the user how much is owed and then prints the minimum number of coins with which that change can be made.

#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(int argc, char **argv)
{
    float dollars;
    int quarters, dimes, nickels, pennies;
        // Execute this until the value is larger than 0
        do
        {
           // Get the dollar amount
           dollars = get_float("Change owed: ");
        }
        while 
            (dollars < 0);
        
    
        int cents = round(dollars*100);
               
        quarters = (cents / 25);
        dimes = ((cents % 25) / 10);
        nickels = (((cents % 25) % 10) / 5);
        pennies = ((((cents % 25) % 10) % 5) / 1);
    
        // Count the number of coins    
        int count = (quarters + dimes + nickels + pennies);
        printf("%i\n", count);
   
}

File: Pset 6/cash/cash.py
# Import the get_float function from cs50 modules
from cs50 import get_float

# Prompt the user to provide a dollar amount
amount = get_float("Change owed: ")

# Check if usage is correct and the user has entered a number
while False:
    amount = get_float("Change owed: ")
if amount < 0:
    amount = get_float("Change owed: ")

# Turn the value in cents:
cents = int(round(amount*100))

# Use modulo function to check for the number of coins from each group
quarters = int((cents / 25))
dimes = int(((cents % 25) / 10))
nickels = int((((cents % 25) % 10) / 5))
pennies = int(((((cents % 25) % 10) % 5) / 1))

# Sum the final number of coins
coins = quarters + dimes + nickels + pennies

# Print the final number of coins owed
print(f"{coins:.0f}")